modules/cco-ccoctl-upgrading.adoc

```diff
@@ -0,0 +1,89 @@
+
+Some `ccoctl` commands make AWS API calls to create or modify AWS resources. To place JSON files on the local file system instead, use the `--dry-run` flag. These JSON files can be reviewed or modified and then applied with the AWS CLI tool using the `--cli-input-json` parameters.
+====
+
+.Prerequisites
+
+* Obtain the {product-title} release image for the version that you are upgrading to.
+
+* Extract and prepare the updated `ccoctl` binary from the release image.
+
+//The test plan has the steps as extract the CredentialsRequest CRs, then extract the ccoctl binary. Does it work to include "Extract and prepare the updated `ccoctl` binary from the release image" as a prereq instead? Makes the task more similar to the install task.
+
+.Procedure
+
+. Extract the list of `CredentialsRequest` objects from the {product-title} release image:
++
+[source,terminal,subs="+quotes"]
+----
+$ oc adm release extract --credentials-requests --cloud=aws --to=__<path_to_directory_with_list_of_credentials_requests>__/credrequests quay.io/__<path_to>__/ocp-release:__<version>__
+----
+
+. Complete appropriate health checks. #elaborate#
+//Test plan has a link to several health checks, but what specifically should the user run and what are they watching out for?
+
+. Use the `ccoctl` tool to process all `CredentialsRequest` objects in the `credrequests` directory:
++
+[source,terminal,subs="+quotes"]
+----
+$ ccoctl aws create-iam-roles --name __<name>__ --region=__<aws_region>__ --credentials-requests-dir=__<path_to_directory_with_list_of_credentials_requests>__/credrequests --identity-provider-arn arn:aws:iam::__<aws_account_id>__:oidc-provider/__<name>__-oidc.s3.__<aws_region>__.amazonaws.com
+----
++
+where:
++
+** _<name>_ is the name used to tag any cloud resources that are created for tracking.
+** _<aws_account_id>_ is the AWS account ID.
+** _<aws_region>_ is the AWS region in which cloud resources will be created.
++
+[NOTE]
+====
+For AWS environments that use alternative IAM API endpoints, such as GovCloud, you must also specify your region with the `--region` parameter.
+====
++
+For each `CredentialsRequest` object, `ccoctl` creates an IAM role with a trust policy that is tied to the specified OIDC identity provider, and a permissions policy as defined in each `CredentialsRequest` object from the {product-title} release image.
+
+. Add any new permissions in the appropriate role and policy on AWS.
+//At this point in the test plan, it is stated that "there are 2 new permissions added for machine-api." How does the user figure this out? Would output from the previous step help?
+//Not really clear on the correct thing to tell the user to do on the AWS side, what is this process called? Any AWS docs we should refer them to?
++
+It may take up to several minutes for the updated permissions to take effect.
```

Comment:
I've seen it take a couple seconds, but never minutes. But if we have reason to state that it make take several minutes, then I guess we leave this more broad caveat in this sentence as-is.